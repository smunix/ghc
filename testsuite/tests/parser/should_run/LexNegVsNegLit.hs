{-# LANGUAGE LexicalNegation, NegativeLiterals #-}

-- LexicalNegation vs NegativeLiterals

data FreeNum
  = FromInteger Integer
  | Negate FreeNum
  deriving (Show)

instance Num FreeNum where
  fromInteger = FromInteger
  negate = Negate

main = do
  let x = 5
  print (-123 :: FreeNum)
  print (-x :: FreeNum)
