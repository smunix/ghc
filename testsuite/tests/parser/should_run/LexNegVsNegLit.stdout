FromInteger (-123)
Negate (FromInteger 5)
